src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2332,10 +2332,33 @@ MergeTreeData::DataPartsVector MergeTreeData::removePartsInRangeFromWorkingSet(c
             throw Exception("Unexpected partition_id of part " + part->name + ". This is a bug.", ErrorCodes::LOGICAL_ERROR);
 
         /// It's a DROP PART and it's already executed by fetching some covering part
-        if (part->info != drop_range && part->info.contains(drop_range))
+        bool is_drop_part = !drop_range.isFakeDropRangePart();
```

Comment:
It also can be `drop partition`. But drop range for partition always have zero `min_block`. Drop range for part also may have zero `min_block`, but such drop range cannot be intersected my merge entry in the way described in the comment. So we can relax check only for some drop ranges for parts, not for partitions.
```suggestion
        bool is_drop_part = !drop_range.isFakeDropRangePart() && drop_range.min_block;
```